File: internal/app/domain/job.go
package domain

import (
	"github.com/google/uuid"
	"time"
)

type Job struct {
	id          uuid.UUID
	url         string
	title       string
	description string
	source      string
	location    string
	remote      bool
	postedAt    time.Time
	createdAt   time.Time
	updatedAt   time.Time
}

type JobOptional func(*Job)

func WithTimestamps(c, u time.Time) JobOptional {
	return func(j *Job) {
		j.createdAt = c
		j.updatedAt = u
	}
}

func NewJob(id uuid.UUID, url, title, description, source, location string, remote bool, postedAt time.Time, opts ...JobOptional) *Job {
	j := &Job{
		id:          id,
		url:         url,
		title:       title,
		description: description,
		source:      source,
		location:    location,
		remote:      remote,
		postedAt:    postedAt,
		createdAt:   time.Now(),
		updatedAt:   time.Now(),
	}

	for _, opt := range opts {
		opt(j)
	}

	return j
}

func (j *Job) ID() uuid.UUID {
	return j.id
}

func (j *Job) URL() string {
	return j.url
}

func (j *Job) Title() string {
	return j.title
}

func (j *Job) Description() string {
	return j.description
}

func (j *Job) Source() string {
	return j.source
}

func (j *Job) Location() string {
	return j.location
}

func (j *Job) Remote() bool {
	return j.remote
}

func (j *Job) PostedAt() time.Time {
	return j.postedAt
}

func (j *Job) CreatedAt() time.Time {
	return j.createdAt
}

func (j *Job) UpdatedAt() time.Time {
	return j.updatedAt
}

File: app/Models/Job.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Carbon;

/**
 * App\Models\Job
 * @property string $id
 * @property string $title
 * @property string $description
 * @property string $source
 * @property string $location
 * @property string $url
 * @property bool $remote
 * @property Carbon $posted_at
 * @property Carbon $created_at
 * @property Carbon $updated_at
 * @property Carbon $deleted_at
 */
class Job extends Model
{
    use HasFactory;

    protected $guarded = [];
}
